### ISSUE TITLE
Bindings do not work in Release build

### ISSUE THREAD

**Issue by takacsalbert** [CodonFramework/Codon#17]
Hi,
I use Codon 1.1.0, and in Debug build everything works fine, but in Release bindings start throwing excepetions like "System.InvalidOperationException: Source property XXX does not exist on XViewModel."
It seems to be only binding to methods crash like: SetImageDrawable, ItemClick. And there is another one: MovementMethod. Binding to properties work though.
Can you please help me out and what is the problem and how can I fix it? Thanks in advance!

**Comment by DVaughan** [comment]
In a Release build it is likely that the Linker is stripping out those properties because it doesn't see that they're being used. If you're using Visual Studio, In the Android Options section of your main project's properties, ensure that the Linking option is set to SDK Assemblies only. If you'd rather include all assemblies in the linking process, you'll need to create a LinkerConfiguration.xml file and exclude those properties in it. Please see the Xamarin documentation on how to do that.

**Comment by takacsalbert** [comment]
Ok, thanks! I'll give it a try. I'll get back when it works.

**Comment by takacsalbert** [comment]
Hi! Still cannot manage it to work. Linking option was already set to SDK Assemblies only, and that causes to remove my code. When I set it to None, everything works fine even in Release build. But of course this is not what I want to do for production.
I've tried several ways to prevent my code from being stripped out:
- [Android.Runtime.Preserve] and [Android.Runtime.Preserve (AllMembers = true)]
- falseflag
- Custom linker config file (followed the example)
- AndroidLinkSkip 

From here:
**https://developer.xamarin.com/guides/android/advanced_topics/linking/**

Nothing seems to help unfortunately.

**Comment by DVaughan** [comment]
I recommend you post a bug report on [Xamarin's Bugzilla site](https://bugzilla.xamarin.com/)

**Comment by takacsalbert** [comment]
Hi!
I was able to solve this problem by adding a LinkerPleaseInclude.cs file. In this file I added methods that are never called, but they tell the Linker to retain the types used there. For example:

public void Include(PageViewModelWrapper wrapper)
        {
            ListView listView = new ListView(null);
            listView.ItemClick += wrapper.HandleItemClick;
        }

Thank you for your guidance! I hope others will find this useful.

Here is a link for reference:
https://github.com/MvvmCross/MvvmCross-Samples/blob/master/Babel/Babel.Droid/LinkerPleaseInclude.cs
